File: 3. Sophomore_HKI/OOP/OOP_CUOIKI/QuanLyShopQuanAo (C#)/src/mypack/SanPham.java
package mypack;

public class SanPham {
	
	public String maSanPham;     // primary key
    public String tenSanPham;

    // foreign key
    public String mathuongHieu;
    public String maPhieuBaoHanh;

    // singular
    public ThuongHieu thuongHieu;
    public PhieuBaoHanh phieuBaoHanh;

	// constructor
    public SanPham() { }
    public SanPham(SanPham sanPham)
    {
        this.maSanPham = sanPham.maSanPham;
        this.tenSanPham = sanPham.tenSanPham;
        this.mathuongHieu = sanPham.mathuongHieu;
        this.maPhieuBaoHanh = sanPham.maPhieuBaoHanh;
        this.thuongHieu = sanPham.thuongHieu;
        this.phieuBaoHanh = sanPham.phieuBaoHanh;
    }
    public SanPham(String maSanPham, String tenSanPham, String mathuongHieu, 
    		String maPhieuBaoHanh,ThuongHieu thuongHieu,PhieuBaoHanh phieuBaoHanh)
    {
        this.maSanPham = maSanPham;
        this.tenSanPham = tenSanPham;
        this.mathuongHieu = mathuongHieu;
        this.maPhieuBaoHanh = maPhieuBaoHanh;
        this.thuongHieu = thuongHieu;
        this.phieuBaoHanh = phieuBaoHanh;
    }
    
    // getter - setter
    public String getMaSanPham() {
		return maSanPham;
	}
	public void setMaSanPham(String maSanPham) {
		this.maSanPham = maSanPham;
	}
	public String getTenSanPham() {
		return tenSanPham;
	}
	public void setTenSanPham(String tenSanPham) {
		this.tenSanPham = tenSanPham;
	}
	public String getMathuongHieu() {
		return mathuongHieu;
	}
	public void setMathuongHieu(String mathuongHieu) {
		this.mathuongHieu = mathuongHieu;
	}
	public String getMaPhieuBaoHanh() {
		return maPhieuBaoHanh;
	}
	public void setMaPhieuBaoHanh(String maPhieuBaoHanh) {
		this.maPhieuBaoHanh = maPhieuBaoHanh;
	}
	public ThuongHieu getThuongHieu() {
		return thuongHieu;
	}
	public void setThuongHieu(ThuongHieu thuongHieu) {
		this.thuongHieu = thuongHieu;
	}
	public PhieuBaoHanh getPhieuBaoHanh() {
		return phieuBaoHanh;
	}
	public void setPhieuBaoHanh(PhieuBaoHanh phieuBaoHanh) {
		this.phieuBaoHanh = phieuBaoHanh;
	}
}

File: 3. Sophomore_HKI/OOP/OOP_CUOIKI/QuanLyShopQuanAo (C#)/SanPham.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyShopQuanAo
{
    public class SanPham
    {
        public string maSanPham { get; set; }     // primary key
        public string tenSanPham { get; set; }

        // foreign key
        public string mathuongHieu { get; set; }
        public string maPhieuBaoHanh { get; set; }

        // singular
        public ThuongHieu thuongHieu { get; set; }
        public PhieuBaoHanh phieuBaoHanh { get; set; }

        // plural
        public ICollection<ChiTietHoaDonNhapHang> dsChiTietNhap_SanPham { get; set; }
        public ICollection<ChiTietHoaDonMuaHang> dsChiTietMua_SanPham { get; set; }

        // constructor
        public SanPham() { }
        public SanPham(SanPham sanPham)
        {
            this.maSanPham = sanPham.maSanPham;
            this.tenSanPham = sanPham.tenSanPham;
            this.mathuongHieu = sanPham.mathuongHieu;
            this.maPhieuBaoHanh = sanPham.maPhieuBaoHanh;
            this.thuongHieu = sanPham.thuongHieu;
            this.phieuBaoHanh = sanPham.phieuBaoHanh;
            this.dsChiTietNhap_SanPham = sanPham.dsChiTietNhap_SanPham;
            this.dsChiTietMua_SanPham = sanPham.dsChiTietMua_SanPham;
        }
        public SanPham(string maSanPham, string tenSanPham, string mathuongHieu, string maPhieuBaoHanh,
            ThuongHieu thuongHieu,PhieuBaoHanh phieuBaoHanh, ICollection<ChiTietHoaDonNhapHang> dsChiTietNhap_SanPham,
            ICollection<ChiTietHoaDonMuaHang> dsChiTietMua_SanPham)
        {
            this.maSanPham = maSanPham;
            this.tenSanPham = tenSanPham;
            this.mathuongHieu = mathuongHieu;
            this.maPhieuBaoHanh = maPhieuBaoHanh;
            this.thuongHieu = thuongHieu;
            this.phieuBaoHanh = phieuBaoHanh;
            this.dsChiTietNhap_SanPham = dsChiTietNhap_SanPham;
            this.dsChiTietMua_SanPham = dsChiTietMua_SanPham;
        }
    }
}
